### PR TITLE
Feature/68116 - cybersource payment method styles

### PR DESCRIPTION
**Pull request by Januszpl**
Result of changes:

image

Plus adding !optional flag for easier spotting which imports can be commented in projects based on this theme. However I can revert it and we can discuss it.

### CODE REVIEW

File: Snowdog_Components/components/Templates/checkout/vendor/CyberSource_SecureAcceptance/_module.scss [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        > .field.date {
+            .select-year {
+                min-width: $cybersource__select-year-min-width;
+            }
+        }
+
+        // reset Snowdog_Blank styles without overriding _select.scss styles
+        .select {
+            padding-left: $cybersource__select-padding-left;
+            padding-right: $cybersource__select-padding-right;
```
[the remark this answers is not in the document]
Comment (Januszpl) [reply Snowdog_Components/components/Templates/checkout/vendor/CyberSource_SecureAcceptance/_module.scss]: yes. done

File: Snowdog_Components/components/Templates/checkout/vendor/CyberSource_SecureAcceptance/_module.scss [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+
+        // hide spinners for number input (cc, ccv fields)
+        input[type="number"]::-webkit-outer-spin-button,
+        input[type="number"]::-webkit-inner-spin-button {
+            -webkit-appearance: none;
+            margin: 0;
+        }
+
+        input[type="number"] {
+            -moz-appearance: textfield;
```
[the remark this answers is not in the document]
Comment (Januszpl) [reply Snowdog_Components/components/Templates/checkout/vendor/CyberSource_SecureAcceptance/_module.scss]: done

File: Snowdog_Components/components/Templates/checkout/vendor/CyberSource_SecureAcceptance/_module.scss [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+        margin-bottom: $cybersource__fieldset-margin-bottom;
+
+        .credit-card-types {
+            justify-content: $cybersource__credit-cards-justify-content;
+        }
+
+        // hide spinners for number input (cc, ccv fields)
+        input[type="number"]::-webkit-outer-spin-button,
+        input[type="number"]::-webkit-inner-spin-button {
+            -webkit-appearance: none;
```
[the remark this answers is not in the document]
Comment (Januszpl) [reply Snowdog_Components/components/Templates/checkout/vendor/CyberSource_SecureAcceptance/_module.scss]: done

File: Snowdog_Components/components/Templates/checkout/vendor/CyberSource_SecureAcceptance/_module.scss [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+.checkout-payment-method {
+    .fieldset {
+        > .field,
+        > .fields > .field {
+            @include lib-form-field();
+        }
+    }
+
+    .payment-method-content fieldset.chcybersource {
+        margin-bottom: $cybersource__fieldset-margin-bottom;
```
[the remark this answers is not in the document]
Comment (Januszpl) [reply Snowdog_Components/components/Templates/checkout/vendor/CyberSource_SecureAcceptance/_module.scss]: done

File: CyberSource_SecureAcceptance/web/template/payment/cc-form.html [hunk 5]
```diff
@@ -0,0 +1,214 @@
...
+    >
+        <label
+            data-bind="attr: {for: getCode() + '_expiration'}"
+            class="label"
+        >
+            <span>
+                <!-- ko i18n: 'Expiration Date'--><!-- /ko -->
+            </span>
+        </label>
+        <div class="control">
```
[the remark this answers is not in the document]
Comment (Januszpl) [reply CyberSource_SecureAcceptance/web/template/payment/cc-form.html:97]: In some places not in some places yes. On shipping step you can see it is everywhere used even tough it's not required there as well. So I think better to keep this structure to have some consistency. If we would remove it here I think we should remove in other places of forms as well.

File: CyberSource_SecureAcceptance/web/template/payment/cc-form.html [hunk 6]
```diff
@@ -0,0 +1,214 @@
...
+    </div>
+    <div
+        class="field date required"
+        data-bind="attr: {id: getCode() + '_cc_type_exp_div'}"
+    >
+        <label
+            data-bind="attr: {for: getCode() + '_expiration'}"
+            class="label"
+        >
+            <span>
```
[the remark this answers is not in the document]
Comment (Januszpl) [reply CyberSource_SecureAcceptance/web/template/payment/cc-form.html]: removed

File: CyberSource_SecureAcceptance/web/template/payment/cc-form.html [hunk 7]
```diff
@@ -0,0 +1,214 @@
...
+    <div class="field number required">
+        <label
+            data-bind="attr: {for: getCode() + '_cc_number'}"
+            class="label"
+        >
+            <span>
+                <!-- ko i18n: 'Credit Card Number'--><!-- /ko -->
+            </span>
+        </label>
+        <div class="control">
```
[the remark this answers is not in the document]
Comment (Januszpl) [reply CyberSource_SecureAcceptance/web/template/payment/cc-form.html:61]: In some places not in some places yes. On shipping step you can see it is everywhere used even tough it is not required there as well. So I think better to keep this structure to have some consistency. If we would remove it here I think we should remove in other places of forms as well.

File: CyberSource_SecureAcceptance/web/template/payment/cc-form.html [hunk 8]
```diff
@@ -0,0 +1,214 @@
...
+        </div>
+    </div>
+    <div class="field number required">
+        <label
+            data-bind="attr: {for: getCode() + '_cc_number'}"
+            class="label"
+        >
+            <span>
+                <!-- ko i18n: 'Credit Card Number'--><!-- /ko -->
+            </span>
```
[the remark this answers is not in the document]
Comment (Januszpl) [reply CyberSource_SecureAcceptance/web/template/payment/cc-form.html]: removed

File: CyberSource_SecureAcceptance/web/template/payment/cc-form.html [hunk 9]
```diff
@@ -0,0 +1,214 @@
...
+                        }"
+                    >
+                    <!--/ko-->
+                </li>
+                <!--/ko-->
+            </ul>
+            <input
+                type="hidden"
+                name="payment[cc_type]"
+                class="input__field"
```
[the remark this answers is not in the document]
Comment (Januszpl) [reply CyberSource_SecureAcceptance/web/template/payment/cc-form.html:42]: removed

File: CyberSource_SecureAcceptance/web/template/payment/cc-form.html (hunk 5 above)
[the remark this answers is not in the document]
Comment (anqaka) [reply CyberSource_SecureAcceptance/web/template/payment/cc-form.html:96]: ok, you're right. thanks for checking

**Review by anqaka — APPROVED** [review]
code looks ok

**Pull request merged**